tests/pubsub/integration/testgossipsubsignatureflags.nim

```diff
@@ -0,0 +1,157 @@
+      receivedMessages[0].fromPeer.data.len > 0
+      receivedMessages[0].seqno.len > 0
+      receivedMessages[0].signature.len > 0
+      receivedMessages[0].key.len > 0
+
+  asyncTest "Sign flag - messages are not signed when sign=false":
+    let nodes = generateNodes(
+      2, gossip = true, sign = false, verifySignature = false, anonymize = false
+    )
+
+    startNodesAndDeferStop(nodes)
+    await connectNodesStar(nodes)
+
+    nodes.subscribeAllNodes(topic, voidTopicHandler)
+
+    var (receivedMessages, checkForMessage) = createCheckForMessages()
+    nodes[1].addOnRecvObserver(checkForMessage)
+
+    tryPublish await nodes[0].publish(topic, testData), 1
+
+    check:
+      receivedMessages[0].data == testData
+      receivedMessages[0].signature.len == 0
+      receivedMessages[0].key.len == 0
+
+  asyncTest "Anonymize flag - messages are anonymous when anonymize=true":
+    let nodes = generateNodes(
+      2, gossip = true, sign = true, verifySignature = true, anonymize = true
+    ) # anonymize = true takes precedence
+    startNodesAndDeferStop(nodes)
+    await connectNodesStar(nodes)
+
+    nodes.subscribeAllNodes(topic, voidTopicHandler)
+
+    var (receivedMessages, checkForMessage) = createCheckForMessages()
+    nodes[1].addOnRecvObserver(checkForMessage)
+
+    let testData = "anonymous message".toBytes()
+    tryPublish await nodes[0].publish(topic, testData), 1
+
+    check:
+      receivedMessages[0].data == testData
+      receivedMessages[0].fromPeer.data.len == 0
+      receivedMessages[0].seqno.len == 0
+      receivedMessages[0].signature.len == 0
+      receivedMessages[0].key.len == 0
+
+  type NodeConfig = tuple[sign: bool, verify: bool, anonymize: bool]
+  let scenarios =
+    @[
+      # (sender_config, receiver_config, should_work)
+
+      # valid combos
+      # S default, R default
+      ((true, true, false), (true, true, false), true),
+      # S default, R anonymous
+      ((true, true, false), (false, false, true), true),
+      # S anonymous, R anonymous
+      ((false, false, true), (false, false, true), true),
+      # S only sign, R only verify
+      ((true, false, false), (false, true, false), true),
+      # S only verify, R only sign
+      ((true, true, true), (false, false, false), true),
+      # S anonymous (not signed despite the flag), R minimal
+      ((false, true, true), (true, false, false), true),
+      # S unsigned, R unsigned
+      ((false, false, false), (false, false, false), true),
+
+      # invalid combos
+      # S anonymous, R default
+      ((false, false, true), (true, true, false), false),
+      # S unsigned, R anonymous but verify 
+      ((false, false, false), (true, true, true), false),
+      # S unsigned, R default
+      ((false, false, false), (true, true, false), false),
+    ]
+
+  for scenario in scenarios:
+    let title = "Compatibility matrix: " & $scenario
+    asyncTest title:
+      let
+        (senderConfig, receiverConfig, shouldWork) = scenario
```

Comment:
would be better if `scenarios` is `seq[Scenario]`, then `type Scenario` has `senderConfig` `receiverConfig` and `shouldWork` properties. `senderConfig` and `receiverConfig` are i guess `NodeConfig` ??